src/MetaMaskInpageProvider.ts

```diff
@@ -506,245 +201,29 @@ export default class MetaMaskInpageProvider extends SafeEventEmitter {
-   * accountsChanged event.
-   *
-   * @param accounts - The new accounts value.
-   * @param isEthAccounts - Whether the accounts value was returned by
-   * a call to eth_accounts.
-   */
-  private _handleAccountsChanged(accounts: unknown[], isEthAccounts = false): void {
-    let _accounts = accounts;
-
-    if (!Array.isArray(accounts)) {
-      this._log.error(
-        'MetaMask: Received invalid accounts parameter. Please report this bug.',
-        accounts,
-      );
-      _accounts = [];
-    }
-
-    for (const account of accounts) {
-      if (typeof account !== 'string') {
-        this._log.error(
-          'MetaMask: Received non-string account. Please report this bug.',
-          accounts,
-        );
-        _accounts = [];
-        break;
-      }
-    }
-
-    // emit accountsChanged if anything about the accounts array has changed
-    if (!dequal(this._state.accounts, _accounts)) {
-
-      // we should always have the correct accounts even before eth_accounts
-      // returns
-      if (isEthAccounts && this._state.accounts !== null) {
-        this._log.error(
-          `MetaMask: 'eth_accounts' unexpectedly updated accounts. Please report this bug.`,
-          _accounts,
-        );
-      }
-
-      this._state.accounts = _accounts as string[];
-
-      // handle selectedAddress
-      if (this.selectedAddress !== _accounts[0]) {
-        this.selectedAddress = _accounts[0] as string || null;
-      }
-
-      // finally, after all state has been updated, emit the event
-      if (this._state.initialized) {
-        this.emit('accountsChanged', _accounts);
       }
     }
   }
 
-  /**
-   * Upon receipt of a new isUnlocked state, sets relevant public state.
-   * Calls the accounts changed handler with the received accounts, or an empty
-   * array.
-   *
-   * Does nothing if the received value is equal to the existing value.
-   * There are no lock/unlock events.
-   *
-   * @param opts - Options bag.
-   * @param opts.accounts - The exposed accounts, if any.
-   * @param opts.isUnlocked - The latest isUnlocked value.
-   */
-  private _handleUnlockStateChanged({
-    accounts,
-    isUnlocked,
-  }: { accounts?: string[]; isUnlocked?: boolean} = {}) {
-    if (typeof isUnlocked !== 'boolean') {
-      this._log.error('MetaMask: Received invalid isUnlocked parameter. Please report this bug.');
-      return;
-    }
-
-    if (isUnlocked !== this._state.isUnlocked) {
-      this._state.isUnlocked = isUnlocked;
-      this._handleAccountsChanged(accounts || []);
-    }
-  }
-
   /**
    * Warns of deprecation for the given event, if applicable.
    */
-  private _warnOfDeprecation(eventName: string): void {
-    if (this._state.sentWarnings.events[eventName as WarningEventName] === false) {
+  protected _warnOfDeprecation(eventName: string): void {
+    if (this.sentWarnings && this.sentWarnings.events[eventName as WarningEventName] === false) {
```

Comment:
having this guard `this.sentWarnings &&` prevents this from running in the constructor, which is not ideal. but typescript initializes its default inherited properties after the base constructor runs

Reply:
Huh, sure, why not!?

I take it we can accomplish the same thing with optional chaining?

```suggestion
    if (this?.sentWarnings.events[eventName as WarningEventName] === false) {
```